File: javascript/api.test.js
const axios = require('axios');

const BASE_URL = 'https://jsonplaceholder.typicode.com';

test('GET /users - should return 200 OK and user list', async () => {
  const response = await axios.get(`${BASE_URL}/users`);
  expect(response.status).toBe(200);
  expect(response.data.length).toBeGreaterThan(0);
  
  const firstUser = response.data[0];
  expect(firstUser).toHaveProperty('name');
  expect(firstUser).toHaveProperty('email');
});

test('POST /users - should return 201 Created', async () => {
  // TODO: Candidate Assignment
  // Write a test to POST a new user to /users.
  // Verify response is 201 and fields match.
});

test('GET /users/9999 - should return 404 Not Found', async () => {
  // TODO: Candidate Assignment
  // Write a negative test requesting /users/9999.
  // Verify response status is 404.
});

File: java/src/test/java/ApiTest.java
import org.junit.jupiter.api.Test;
import static io.restassured.RestAssured.*;
import static org.hamcrest.Matchers.*;

public class ApiTest {
    private static final String BASE_URL = "https://jsonplaceholder.typicode.com";

    @Test
    public void testGetUsersShouldReturn200() {
        given()
            .baseUri(BASE_URL)
        .when()
            .get("/users")
        .then()
            .statusCode(200)
            .body("size()", greaterThan(0))
            .body("[0].name", notNullValue())
            .body("[0].email", notNullValue());
    }

    @Test
    public void testCreateUserShouldReturn201() {
        // TODO: Candidate Assignment
        // Write a test to POST a new user to /users.
        // Verify response is 201 Created and response mirrors body fields.
    }

    @Test
    public void testGetNonExistentUserShouldReturn404() {
        // TODO: Candidate Assignment
        // Write a negative test requesting /users/9999.
        // Verify code returns 404 Not Found.
    }
}
